File: Medium/2 Add Two Numbers/Solution.py
# https://leetcode.com/problems/add-two-numbers/submissions/689827814/

# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def getSum(self, node):
        current = node
        ans = str(current.val)
        while current.next is not None:
            current = current.next
            ans = str(current.val) + ans
        return ans
    
    def nodeify(self, number):
        next = None
        current = ListNode(val=int(number[0]),next=next)
        for i in range(1,len(number)):
            next = current
            current = ListNode(val=int(number[i]),next=next)
        return current
    
    def addTwoNumbers(self, l1, l2):
        """
        :type l1: ListNode
        :type l2: ListNode
        :rtype: ListNode
        """
        
        return self.nodeify(str(int(self.getSum(l1))+int(self.getSum(l2))))

File: Medium/2 Add Two Numbers/Solution.java
// https://leetcode.com/problems/add-two-numbers/submissions/689817784/

import java.math.BigInteger;
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    
    private String getSum(ListNode node){
        ListNode current = node;
        String sum = Integer.toString(current.val);
        while (current.next != null){
            current = current.next;
            sum = Integer.toString(current.val) + sum;
            System.out.println("Current sum is: " + sum);
        }
        return sum;
    }
    
    private ListNode nodeify(String str){
        ListNode next = null;
        ListNode current = new ListNode(Character.getNumericValue(str.charAt(0)), next);
        for (int i = 1; i < str.length(); i++) {
            next = current;
            current = new ListNode(Character.getNumericValue(str.charAt(i)), next);
        }
        return current;
        
    }
    
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        BigInteger num1 = new BigInteger(getSum(l1));
        BigInteger num2 = new BigInteger(getSum(l2));
        System.out.println(num1);
        System.out.println(num2);
        BigInteger sum = num1.add(num2);
        System.out.println(sum);
        return nodeify(sum.toString());
    }
}